### ISSUE TITLE
Major Security Issue - Forgot password

### ISSUE THREAD

**Issue by samuelkilada** [commercetools/sunrise-spa#296]
Hello. I don't know if I'm missing something, but I noticed that in this file it creates a reset password token:

https://github.com/commercetools/sunrise-spa/blob/a556c17e158d8beb02d5d14b6c70b7ec114e03e7/src/presentation/fashion/Login/ForgotPassword/ForgotPassword.js

`const createToken = () => {
      return createResetToken(form.value.email).then(
        (result) =>
          gotoResetToken(
            result.data.customerCreatePasswordResetToken
              .value
          )
      );
};`

If you look at the gotoResetToken function, it looks like this:

https://github.com/commercetools/sunrise-spa/blob/fc2f51e25ffa0c9ab05b5ec545ca1221342dc691/composition/useCustomerTools.js

`const gotoResetToken = (token) =>
    router.push({
      name: 'reset-password',
      params: { token },
    });`

In other words, it's generating a reset token and then redirecting the user to the reset password page with that token! That means you could enter anyone's email address and reset their password with it. Am I missing something here?

**Comment by harm-meijer** [comment]
@samuelkilada You are correct, the [reset password](https://github.com/commercetools/sunrise-spa/tree/development/Docs#forgot-password) works like this for demo only. In production you will need a proxy to handle creating the token and email it to the user.

**Comment by samuelkilada** [comment]
@harm-meijer Thanks very much for the quick reply. I would highly recommend adding a 'todo" statement or some kind of security warning because there may be many developers who end up implementing this feature in this way without realizing the security implications.

**Comment by harm-meijer** [comment]
@samuelkilada If you create the api client with mobile and web application scope as per [the instructions](https://github.com/commercetools/sunrise-spa/tree/development/Docs#Environment-variables) then reset password [link](https://github.com/commercetools/sunrise-spa/blob/development/src/presentation/fashion/Login/LoginForm/LoginForm.js#L28) would not be [available](https://github.com/commercetools/sunrise-spa/blob/development/composition/useAccessRules.js#L35-L46).

I will add a comment in the forgot [password files](https://github.com/commercetools/sunrise-spa/tree/development/src/presentation/fashion/Login/ForgotPassword) as well.

**Comment by samuelkilada** [comment]
@harm-meijer That makes sense. Thanks for doing that!
